ui/pages/password-reset/index.js

```diff
@@ -0,0 +1,185 @@
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { useSWRConfig } from 'swr'
+
+import LoginLayout from '../../components/layouts/login'
+
+export default function PasswordReset() {
+  const { mutate } = useSWRConfig()
+  const router = useRouter()
+  const { token } = router.query
+
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState('')
+  const [submitted, setSubmitted] = useState(false)
+
+  async function onSubmitSetPassword(e) {
+    e.preventDefault()
+
+    try {
+      const res = await fetch('/api/password-reset', {
+        method: 'post',
+        body: JSON.stringify({
+          token,
+          password,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+
+      await mutate('/api/users/self')
+      router.replace('/')
+    } catch (e) {
+      console.error(e)
+    }
+
+    return false
+  }
+
+  async function onSubmit(e) {
+    e.preventDefault()
+
+    try {
+      const res = await fetch('/api/password-reset-request', {
+        method: 'post',
+        body: JSON.stringify({
+          email,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+      setSubmitted(true)
+    } catch (e) {
+      console.error(e)
```

Comment:
do we want to put the the error `e` to setError? so it can be displayed on the UI?